### PR TITLE
Added DataprepGetJobsForJobGroupOperator

### PR DESCRIPTION
**Pull request by michalslowikowski00**
This is task is about adding new operator in Airflow -- Dataprerp.
Dataprep is cloud data service for analysis and machine learning.

related: https://github.com/apache/airflow/issues/

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/dataprep.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    """
+
+    def __init__(self, dataprep_conn_id: str = "dataprep_conn_id") -> None:
+        super().__init__()
+        self.dataprep_conn_id = dataprep_conn_id
+        self._url = "https://api.clouddataprep.com/v4/jobGroups"
+
+    @property
+    def _headers(self):
```
Comment (potiuk) [inline airflow/providers/google/cloud/hooks/dataprep.py]:
```suggestion
    def _headers(self) -> Dict[str, str]:
```

File: airflow/providers/google/cloud/hooks/dataprep.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    def get_jobs_for_job_group(self, job_id: int):
+        """
+        Get information about the batch jobs within a Cloud Dataprep job.
+
+        :param job_id The ID of the job that will be fetched.
+        :type job_id: int
+        """
+        url: str = f"{self._url}/{job_id}/jobs"
+        response = requests.get(url, headers=self._headers)
+        response.raise_for_status()
```
Comment (potiuk) [inline airflow/providers/google/cloud/hooks/dataprep.py:74]: Nice! I did not know it existed  :)

File: docs/howto/operator/google/cloud/dataprep.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Google Dataprep Operators
+=======================================
```
Comment (potiuk) [inline docs/howto/operator/google/cloud/dataprep.rst]: Too long "underline' (====) -> should be the same length as the text above.

File: tests/providers/google/cloud/hooks/test_dataprep.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        side_effect=[mock.MagicMock(), HTTPError()],
+    )
+    def test_should_not_retry_after_success(self, mock_get_request, mock_hook):
+        mock_hook.get_jobs_for_job_group.retry.sleep = mock.Mock()  # pylint: disable=no-member
+        mock_hook.get_jobs_for_job_group(JOB_ID)
+        assert mock_get_request.call_count == 1
+
+    @patch(
+        "airflow.providers.google.cloud.hooks.dataprep.requests.get",
+        side_effect=[
```
Comment (potiuk) [inline tests/providers/google/cloud/hooks/test_dataprep.py:75]: Nice test!

File: tests/providers/google/cloud/hooks/test_dataprep.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    def test_should_retry_after_four_errors(self, mock_get_request, mock_hook):
+        mock_hook.get_jobs_for_job_group.retry.sleep = mock.Mock()  # pylint: disable=no-member
+        mock_hook.get_jobs_for_job_group(JOB_ID)
+        assert mock_get_request.call_count == 5
+
+    @patch(
+        "airflow.providers.google.cloud.hooks.dataprep.requests.get",
+        side_effect=[HTTPError(), HTTPError(), HTTPError(), HTTPError(), HTTPError()],
+    )
+    def test_raise_error_after_five_calls(self, mock_get_request, mock_hook):
```
Comment (potiuk) [inline tests/providers/google/cloud/hooks/test_dataprep.py:92]: Should  we mention the number of retries and exponential back-off in the docs somewhere?  I am not sure about that one. I think we are not mentioning it elsewhere (assuming that the hard-coded values are reasonable). But maybe (possibly in a separate PR) we could make it more explicit on the typical retry scenarios we use for GCP classes? Or maybe we should make it somehow configurable for all GCP classes ? WDYT @turbaszek @mik-laj ?

**Review by potiuk — COMMENTED** [review]
Few nits and questions. Good Job @michalslowikowski00 !

File: airflow/providers/google/cloud/hooks/dataprep.py (hunk 2 above)
Comment (michalslowikowski00) [reply airflow/providers/google/cloud/hooks/dataprep.py:74]: Thank you. :)
It is very handy simple method, isn't it?

File: docs/howto/operator/google/cloud/dataprep.rst (hunk 3 above)
Comment (michalslowikowski00) [reply docs/howto/operator/google/cloud/dataprep.rst]: Aw... yes indeed. Interesting, `./breeze build-docs` did not return an error. Thanks for pointing it out.

File: airflow/providers/google/cloud/hooks/dataprep.py (hunk 1 above)
Comment (michalslowikowski00) [reply airflow/providers/google/cloud/hooks/dataprep.py]: Fixed.

File: tests/providers/google/cloud/hooks/test_dataprep.py (hunk 4 above)
Comment (michalslowikowski00) [reply tests/providers/google/cloud/hooks/test_dataprep.py:75]: :)

File: tests/providers/google/cloud/hooks/test_dataprep.py (hunk 5 above)
Comment (michalslowikowski00) [reply tests/providers/google/cloud/hooks/test_dataprep.py:92]: I did not mention anywhere else that tenacity retry has 5 calls. This case is only my interpretation. I was suggesting how it was done already done in airflow -- I mean that `num_retries`. I am open for suggestions.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Thanks @michalslowikowski00 !

File: tests/providers/google/cloud/hooks/test_dataprep.py (hunk 5 above)
Comment (turbaszek) [reply tests/providers/google/cloud/hooks/test_dataprep.py:92]: +1 for making the number of retries configurable

**Pull request merged**